tests/Control/UserDefinedFormControllerTest.php

```diff
@@ -367,4 +370,58 @@ public function testRecipientSubjectMergeFields()
         // check emails
         $this->assertEmailSent('test@example.com', 'no-reply@example.com', 'Email Subject: Basic Value');
     }
+
+    public function testConfirmfolderformInvalidRequest() {
+        $this->logInWithPermission('CMS_ACCESS_CMSMain');
+
+        $url = 'UserDefinedFormController/confirmfolderform?';
+        $userDefinedFormID = $this->idFromFixture(UserDefinedForm::class, 'basic-form-page');
+        $fieldID = $this->idFromFixture(EditableFileField::class, 'fileuploadfield');
+
+
+        $response = $this->get($url . http_build_query(['UserFormID' => $userDefinedFormID]));
+        $this->assertEquals(400, $response->getStatusCode(), 'Request without ID parameter is invalid');
+
+        $response = $this->get($url . http_build_query(['ID' => $fieldID]));
+        $this->assertEquals(400, $response->getStatusCode(), 'Request without UserFormID parameter is invalid');
+
+        $response = $this->get($url . http_build_query(['ID' => $fieldID, 'UserFormID' => -1]));
+        $this->assertEquals(400, $response->getStatusCode(), 'Request with unknown UserFormID is invalid');
+
+        $response = $this->get($url . http_build_query(['ID' => -1, 'UserFormID' => $userDefinedFormID]));
+        $this->assertEquals(200, $response->getStatusCode(), 'Request with unknown ID and known UserFormID is valid');
+    }
+
+    public function testConfirmfolderformAccessControl() {
```

Comment:
What would PSR-2 say about not putting those braces on new lines?

o͡͡͡╮[ ⊡ 〜 ⊡ ]╭o͡͡͡